homeassistant/config_entries.py

```diff
@@ -3485,7 +3485,22 @@ async def async_finish_flow(
         entry = self.hass.config_entries.async_get_known_entry(flow.handler)
 
         if result["data"] is not None:
-            self.hass.config_entries.async_update_entry(entry, options=result["data"])
+            automatic_reload = False
+            if isinstance(flow, OptionsFlowWithReload):
+                automatic_reload = flow.automatic_reload
+
+            if automatic_reload and entry.update_listeners:
+                raise ValueError(
```

Comment:
This will also be thrown, when an integration implements the `OptionsFlowWithReload` and a reconfiguration step (_which still needs the `entry.update_listeners`_)

Reply:
There is no reconfigure step inside an options flow. This is only option flows which should always end with `create_entry` hence the handling here to do the restart instead of using update listeners.

Reply:
Reading discord I understood what you mean. Yes, you need to choose between this new class or using update listeners. You can't have both. So a `reconfigure` flow should use `async_update_reload_and_abort` instead to finalize.

Reply:
I'll have a look into `async_update_reload_and_abort` , thx for that hint 👍